### ISSUE TITLE
code_generation routing: drop non-Gemma shadows; Gemma-only fleet for DPO

### ISSUE THREAD

**Issue by gotoplanb** [gotoplanb/conduct#40]
Per [wander bench v2](https://github.com/gotoplanb/wander/blob/main/bench/reports/v2/report.md): on 10 differential algorithm specs across the post-[conduct#37](https://github.com/gotoplanb/conduct/issues/37) `code_generation` fleet, the non-Gemma shadows are not competitive and are now actively harmful to the upcoming DPO run.

## Bench result (v2)

| Model | Kind | Compile rate | Golden 5/5 | Avg golden |
|---|---|---:|---:|---:|
| `gemma4:e4b` | **primary** | **80%** | **70%** | **3.70** |
| `gemma4:12b` | shadow | 40%¹ | 40% | 2.00 |
| `llama3.2:3b` | shadow | 30% | 20% | 1.10 |
| `qwen3.5:9b` | shadow | 10% | 10% | 0.50 |

¹ `gemma4:12b`'s low compile rate in v2 was an artifact of the fence-parser bug fixed in [conduct#38](https://github.com/gotoplanb/conduct/issues/38), not model quality — on the 4 specs where it did compile it scored 5/5 golden every time. v3 will measure it cleanly.

## Ask

Two changes to the `code_generation` rule in [`conduct/config/seed.routing.yaml`](https://github.com/gotoplanb/conduct/blob/main/config/seed.routing.yaml):

1. **Remove `qwen3.5:9b` from `eval_shadow_models`** — 10% compile, 0.5 avg golden. Not signal.
2. **Remove `llama3.2:3b` from `eval_shadow_models`** — 30% compile, 1.1 avg golden. Different model family / tokenizer, so its preference pairs would pollute Gemma DPO training data.

```yaml
  - task_type: code_generation
    preferred_model: gemma4:e4b           # unchanged
    fallback_model: claude-opus-4-5       # unchanged
    sensitivity: public
    eval_shadow_models:
      - { model: gemma4:12b, rate: 1.0 }  # the ceiling we're chasing
      # qwen3.5:9b removed — non-competitive
      # llama3.2:3b removed — non-competitive + wrong family for DPO
```

**Optional addition** — if a smaller Gemma is available locally (`gemma4:2b`, `gemma3:1b`, whatever the current floor is), adding it as a third shadow would give us the full Gemma scaling curve (floor → e4b → 12b → fine-tuned-e4b) for the lift report. Not required for v3 to be useful — just nice for the narrative.

## Why this matters now

The next milestone is [wander#7](https://github.com/gotoplanb/wander/issues/7) (DPO fine-tune of `gemma4:e4b` toward `gemma4:12b`-like outputs). The research question is *\"can fine-tuned small-Gemma close the gap to big-Gemma?\"* — that requires:

1. **Same-family preference pairs.** DPO is sensitive to instruction-following format and tokenizer quirks. Mixing in qwen/llama \"rejected\" examples teaches the model to avoid *their* failure modes, not e4b's.
2. **A clean ceiling measurement.** With conduct#38 fixed, `gemma4:12b` finally gives us a real target to chase.
3. **Less eval noise.** v2 burned ~60% of its compile-time budget on shadows that never compiled. Trimming to Gemma-only cuts wall-clock and lets us run the full 20-spec suite (vs the halved 10) without context burn.

## What we get out of it

- Clean Gemma scaling-curve measurement in v3.
- DPO export ([conduct#31](https://github.com/gotoplanb/conduct/issues/31), `method=composite`) produces pairs where \"chosen vs rejected\" reflects *capability*, not *prompt-format compatibility*.
- The non-Gemma shadows' history stays in `runs.db` (v1 + v2 reports keep their numbers) — we're not erasing the comparison, just retiring it.

## Related

- [wander bench v2](https://github.com/gotoplanb/wander/blob/main/bench/reports/v2/report.md) — source data
- [conduct#37](https://github.com/gotoplanb/conduct/issues/37) — prior routing change (primary swap + 12b shadow add)
- [conduct#38](https://github.com/gotoplanb/conduct/issues/38) — parser fix that unblocks clean 12b measurement
- [wander#7](https://github.com/gotoplanb/wander/issues/7) — DPO run, beneficiary of this change

🤖 Filed via [Claude Code](https://claude.com/claude-code)

**Comment by gotoplanb** [comment]
## Done — 36ae046 (config, live + persisted)

`code_generation` is now a **Gemma-only fleet**: primary `gemma4:e4b`, shadow `gemma4:12b`. Dropped `qwen3.5:9b` + `llama3.2:3b` per bench v2. The same-family DPO argument is right — keeping cross-family rejected examples would train against *their* failure modes, not e4b's.

```yaml
code_generation:
  preferred_model: gemma4:e4b
  fallback_model: gemma4:12b   # kept LOCAL (see note)
  eval_shadow_models:
    - gemma4:12b   # the ceiling; same family for clean DPO pairs
```

Applied live (`PUT /routing`) + persisted in `seed.routing.yaml`. Routing-YAML test + lint green.

### Two notes
- **Fallback stays `gemma4:12b`, not `claude-opus-4-5`.** Your YAML marked opus "unchanged," but the live fallback has been local `gemma4:12b` since #37 (I keep code-gen fully local — the build sandbox is local-only, never cloud). Same deliberate departure I flagged in #37. Say the word if you want a cloud ceiling fallback for the bench and I'll add it.
- **Smaller-Gemma (optional scaling-curve floor):** I didn't add one — I don't want to guess a tag that isn't actually pulled in this Ollama instance (stale-recall risk). If you give me the exact tag that's installed (e.g. confirm `gemma3:1b` / `gemma4:2b` via `ollama list`), I'll add it as a third shadow for the floor→e4b→12b→fine-tuned curve.

v3 now runs a lean Gemma-only suite — less compile-budget burn, clean ceiling, same-family `method=composite` pairs for wander#7. Closing.
